### PR TITLE
Updates pckages and jest for the tests

### PR DESCRIPTION
**Pull request by joker-777**
* uses now `jest` for the tests which doesn't need webpack. Instead, I added a mall preprocessor `test/preprocessor.js"` to convert the coffee files
* converted all test files for jest
* upgraded all packages including webpack 5
* added `webpack-bundle-analyzer` and created a script to analyze the packages
* simplified the scripts in `package.json`. I don't think there is no need to create a development version. I checked other packages and it isn't common to create the min file. The source map is now created as a separate file too. Happy to discuss this. Maybe I'm wrong...

### CODE REVIEW

**Comment by gingerlime** [comment]
@joker-777 thank you! Regarding the `min` file, I was creating it so people without npm can simply download the script and use it with `<script src="...">`. This is embedded in lamed for example. See https://asciinema.org/a/316783?speed=2&t=0:47

(actually being able to require / import it was added later as a secondary use case)

File: src/adapters.coffee [hunk 1]
```diff
@@ -177,7 +177,7 @@ class Adapters
       utils.sha1("#{@namespace}.#{experiment.name}.#{experiment.user_id}")
 
     _track: (experiment, variant, goal) ->
-      utils.log("Persistent Queue Keen track: #{experiment.name}, #{variant}, #{event}")
+      utils.log("Persistent Queue Keen track: #{experiment.name}, #{variant}, #{goal.name}")
```
Comment (gingerlime) [inline src/adapters.coffee:180]: actually I think we can remove the keen adapter. If anyone uses it, they can use their own adapter!

File: src/utils.coffee [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
+import options from './options'
+
+sha1 = require('sha1')
 
 class Utils
-  @defaults: _.defaults
-  @keys: _.keys
-  @remove: _.remove
-  @omit: _.omit
+  @defaults: (obj, defaults) ->
```
Comment (gingerlime) [inline src/utils.coffee:7]: is there a simple way to test these against lodash's methods to make sure they are compatible?

File: test/utils_test.coffee [hunk 3]
```diff
@@ -1,17 +1,28 @@
-test = require('tape')
-sinon = require('sinon')
-_ = require('lodash')
-Utils = require('../src/utils')
+import Utils from '../src/utils'
+import _ from 'lodash'
 
-setup = ->
-  utils = new Utils()
+test 'omit', () ->
```
Comment (gingerlime) [inline test/utils_test.coffee:4]: ah, nice!

File: src/adapters.coffee (hunk 1 above)
Comment (joker-777) [reply src/adapters.coffee:180]: Do you want me to remove it?

**Comment by joker-777** [comment]
Yes, actually `kenhub.js` is the old `kenhub.min.js`

File: src/adapters.coffee (hunk 1 above)
Comment (gingerlime) [reply src/adapters.coffee:180]: we can do it separately. just thought it’s something that can make Alephbet smaller

**Comment by gingerlime** [comment]
> Yes, actually `kenhub.js` is the old `kenhub.min.js`

Do you mean alephbet.js ? but if it’s minified then I’d like to keep the name so anyone who uses the link can continue to use it and get the latest minified version

**Comment by gingerlime** [comment]
I'm trying to run tests inside docker and get these failures

```
npm run test

> alephbet@0.20.1 test /home/circleci/project
> jest --verbose

 FAIL  test/alephbet_test.coffee
  ● Test suite failed to run

    Cannot find module 'request'
    Require stack:
    - /home/circleci/project/node_modules/request-promise-native/lib/rp.js
    - /home/circleci/project/node_modules/jsdom/lib/jsdom/browser/resources/resource-loader.js
    - /home/circleci/project/node_modules/jsdom/lib/api.js
    - /home/circleci/project/node_modules/jest-environment-jsdom/build/index.js
    - /home/circleci/project/node_modules/jest-runner/build/runTest.js
    - /home/circleci/project/node_modules/jest-runner/build/index.js
    - /home/circleci/project/node_modules/@jest/core/build/TestScheduler.js
    - /home/circleci/project/node_modules/@jest/core/build/jest.js
    - /home/circleci/project/node_modules/jest/node_modules/jest-cli/build/cli/index.js
    - /home/circleci/project/node_modules/jest/node_modules/jest-cli/bin/jest.js
    - /home/circleci/project/node_modules/jest/bin/jest.js

      at node_modules/request-promise-native/lib/rp.js:8:12
      at module.exports (node_modules/stealthy-require/lib/index.js:62:23)
      at Object.<anonymous> (node_modules/request-promise-native/lib/rp.js:7:15)

 FAIL  test/adapters_test.coffee
  ● Test suite failed to run

    Cannot find module 'request'
    Require stack:
    - /home/circleci/project/node_modules/request-promise-native/lib/rp.js
    - /home/circleci/project/node_modules/jsdom/lib/jsdom/browser/resources/resource-loader.js
    - /home/circleci/project/node_modules/jsdom/lib/api.js
    - /home/circleci/project/node_modules/jest-environment-jsdom/build/index.js
    - /home/circleci/project/node_modules/jest-runner/build/runTest.js
    - /home/circleci/project/node_modules/jest-runner/build/index.js
    - /home/circleci/project/node_modules/@jest/core/build/TestScheduler.js
    - /home/circleci/project/node_modules/@jest/core/build/jest.js
    - /home/circleci/project/node_modules/jest/node_modules/jest-cli/build/cli/index.js
    - /home/circleci/project/node_modules/jest/node_modules/jest-cli/bin/jest.js
    - /home/circleci/project/node_modules/jest/bin/jest.js

      at node_modules/request-promise-native/lib/rp.js:8:12
      at module.exports (node_modules/stealthy-require/lib/index.js:62:23)
      at Object.<anonymous> (node_modules/request-promise-native/lib/rp.js:7:15)

 FAIL  test/utils_test.coffee
  ● Test suite failed to run

    Cannot find module 'request'
    Require stack:
    - /home/circleci/project/node_modules/request-promise-native/lib/rp.js
    - /home/circleci/project/node_modules/jsdom/lib/jsdom/browser/resources/resource-loader.js
    - /home/circleci/project/node_modules/jsdom/lib/api.js
    - /home/circleci/project/node_modules/jest-environment-jsdom/build/index.js
    - /home/circleci/project/node_modules/jest-runner/build/runTest.js
    - /home/circleci/project/node_modules/jest-runner/build/index.js
    - /home/circleci/project/node_modules/@jest/core/build/TestScheduler.js
    - /home/circleci/project/node_modules/@jest/core/build/jest.js
    - /home/circleci/project/node_modules/jest/node_modules/jest-cli/build/cli/index.js
    - /home/circleci/project/node_modules/jest/node_modules/jest-cli/bin/jest.js
    - /home/circleci/project/node_modules/jest/bin/jest.js

      at node_modules/request-promise-native/lib/rp.js:8:12
      at module.exports (node_modules/stealthy-require/lib/index.js:62:23)
      at Object.<anonymous> (node_modules/request-promise-native/lib/rp.js:7:15)

Test Suites: 3 failed, 3 total
Tests:       0 total
Snapshots:   0 total
Time:        33.485 s
Ran all test suites.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! alephbet@0.20.1 test: `jest --verbose`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the alephbet@0.20.1 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/circleci/.npm/_logs/2020-12-10T16_51_04_969Z-debug.log
```

side-note: I want to set up some kind of a CI for this. Maybe Github actions? (never used before) or Sempahore...

**Comment by joker-777** [comment]
Ah, I think we really have to introduce yarn finally.

**Comment by gingerlime** [comment]
Yep, another great idea 👍

**Comment by joker-777** [comment]
@gingerlime I just pushed yarn. It should work now. Maybe do `rm -rf node_modules` before running `yarn install`.

**Comment by gingerlime** [comment]
@joker-777 I tried to add github actions, but somehow it fails, can you check? 

https://github.com/Alephbet/alephbet/commit/72d7688e72baea18267a39afda52a4a9205d2f4a

I also need to figure out how to push stuff directly to this PR ;-)

**Comment by joker-777** [comment]
Hmm, no idea. I will check.

**Comment by joker-777** [comment]
I have the feeling that the worflow job only gets triggered once it is merged. On my repository I can see the jobs and they run successfully. I'm not sure how I can give you access to my forked repo...
Screenshot_20201210_190406

**Comment by gingerlime** [comment]
Strange... anyway, I'll merge it, and then anyway I need to tag and publish it separately

**Comment by gingerlime** [comment]
@joker-777 when I try to build it, I get an error

```
$ yarn build
yarn run v1.22.0
$ webpack
[webpack-cli] Invalid configuration object. Webpack has been initialized using a configuration object that does not match the API schema.
 - configuration.node has an unknown property 'fs'. These properties are valid:
   object { __dirname?, __filename?, global? }
   -> Options object for node compatibility features.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by gingerlime** [comment]
maybe it's a local issue? seems to run on github https://github.com/Alephbet/alephbet/pull/34

**Comment by joker-777** [comment]
How were you able to resolve this issue?

**Comment by gingerlime** [comment]
It turned out to be a local old webpack.config.js file that got loaded instead of the coffee file

**Pull request merged**